### PR TITLE
[URDF] Move parser into the library.

### PR DESCRIPTION
**Pull request by jmirabel**
Solves #1078 and #1089

### CODE REVIEW

**Comment by jmirabel** [comment]
Note that those changes may not be backward compatible. It will break the code of dependencies relying on Pinocchio's preprocessor variable related to URDFDOM.

**Comment by jmirabel** [comment]
e.g. see humanoid-path-planner/hpp-pinocchio@d33104c057aa041efd12042fb7af414243957771

**Comment by jcarpent** [comment]
> Note that those changes are not be backward compatible. It will break the code of dependencies relying on Pinocchio's preprocessor variable related to URDFDOM.

I don't really get the point. Can you explain more?

**Comment by jmirabel** [comment]
`PINOCCHIO_URDF_SHARED_PTR` is not available anymore.

File: src/parsers/urdf/utils.hpp [hunk 1]
```diff
@@ -1,10 +1,10 @@
 //
-// Copyright (c) 2015-2016 CNRS
+// Copyright (c) 2015-2020 CNRS INRIA
 //
 
 #ifndef __pinocchio_parsers_urdf_utils_hpp__
 #define __pinocchio_parsers_urdf_utils_hpp__
 
-#warning "This file is not deprecated and will be removed in future releases. There is no replacement."
+#warning "This file is now deprecated and will be removed in future releases. There is no replacement."
```
Comment (jmirabel) [inline src/parsers/urdf/utils.hpp]: Thanks !

**Comment by jmirabel** [comment]
Not related: I suggest we make it clear what should be considered API and what is internal.
- currently, there are the two namespaces *internal* and *details*. One of them is sufficient.
- for C++ macros, we can use a Doxygen macro to create a list of stable macros and a list of internal macros.

**Comment by jcarpent** [comment]
I think internal naming is better.
Do agree for Doxygen too.

**Pull request merged**